### ISSUE TITLE
Add `outputFiles` output property in a structure {filename: internal_storage_uri} to the s3.Downloads task

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/plugin-aws#396]
### Feature description

Suggestion: add `outputFiles` output property in a structure {filename: internal_storage_uri} to all Download tasks so they can be easily passed as `inputFiles` to all script tasks 

Example `outputFiles` output for S3 Downloads task:

```
{"sales/customers.csv":"kestra:///myteam/myflow/executions/6rv7hd4MElMROlwlmkwbqQ/tasks/get/6n7ATc5Lpx6fxULVfxvl3H/16707808184443030721.csv",
"sales/orderlines.csv":"kestra:///myteam/myflow/executions/6rv7hd4MElMROlwlmkwbqQ/tasks/get/6n7ATc5Lpx6fxULVfxvl3H/11267904361238013311.csv",
"sales/products.csv":"kestra:///myteam/myflow/executions/6rv7hd4MElMROlwlmkwbqQ/tasks/get/6n7ATc5Lpx6fxULVfxvl3H/7295630879705972620.csv"}
```

Reason: it's painful to pass downloaded files to script tasks, you need to use really complex pebble:

`inputFiles: "{{ outputs.downloadsOldData.objects | jq('reduce .[] as $item ({}; .[$item.key] = $item.uri)') | first }}"`

full reproducer:

```yaml
id: myflow
namespace: myteam
description: Save and Execute the flow

tasks:
  - id: downloadsOldData
    type: io.kestra.plugin.aws.s3.Downloads
    accessKeyId: aaa
    secretKeyId: aaa
    region: "eu-central-1"
    bucket: annageller
    prefix: sales/
    action: NONE
  
  - id: get
    type: io.kestra.core.tasks.debugs.Return
    format: "{{ outputs.downloadsOldData.objects | jq('reduce .[] as $item ({}; .[$item.key] = $item.uri)') | first }}"

  - id: mergeData
    inputFiles: "{{ outputs.downloadsOldData.objects | jq('reduce .[] as $item ({}; .[$item.key] = $item.uri)') | first }}"
    type: io.kestra.plugin.scripts.shell.Commands
    runner: PROCESS
    commands:
      - ls -R .
```

It seems easy to add new property `outputFiles` to avoid this pain

**Comment by anna-geller** [comment]
@loicmathieu this one doesn't fully work yet as the directory is currently included in the output:

![image](https://github.com/kestra-io/plugin-aws/assets/86264395/f214600d-da0a-4527-ab69-1d3e477537d7)

```yaml
id: process_files
namespace: dev

tasks:
  - id: download
    type: io.kestra.plugin.aws.s3.Downloads
    region: us-east-1
    bucket: kestra-us
    prefix: sales/
    action: NONE

  - id: transform
    inputFiles: "{{ outputs.download.outputFiles }}"
    type: io.kestra.plugin.scripts.shell.Commands
    taskRunner:
      type: io.kestra.plugin.core.runner.Process
    commands:
      - ls -R .
```

this will result in:

```Java
java.io.FileNotFoundException: /var/folders/x_/d0prmph15sd55qs2lbww21740000gn/T/1P9OXvacBVdW0f2sjWHcaJ/sales (Is a directory)
	at java.base/java.io.FileOutputStream.open0(Native Method)
	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:289)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:230)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:179)
	at io.kestra.core.runners.FilesService.lambda$inputFiles$0(FilesService.java:53)
	at io.kestra.core.utils.Rethrow.lambda$throwBiConsumer$1(Rethrow.java:60)
	at java.base/java.util.HashMap.forEach(HashMap.java:1429)
	at io.kestra.core.runners.FilesService.inputFiles(FilesService.java:39)
	at io.kestra.plugin.scripts.exec.scripts.runners.CommandsWrapper.run(CommandsWrapper.java:146)
	at io.kestra.plugin.scripts.shell.Commands.run(Commands.java:134)
	at io.kestra.plugin.scripts.shell.Commands.run(Commands.java:18)
	at io.kestra.core.runners.WorkerTaskThread.doRun(WorkerTaskThread.java:77)
	at io.kestra.core.runners.AbstractWorkerThread.run(AbstractWorkerThread.java:56)
```

**TL;DR we need to exclude the directory itself from the output** and only include the paths to files

let's cross check that we don't include folders in the `outputFiles` output on all `Downloads` tasks

**Comment by loicmathieu** [comment]
Directory has been filtered in https://github.com/kestra-io/plugin-aws/commit/cd69066f97d215e2b6adb8ecd8a729ee846e19a3

I'll check the other Download tasks
